meshtastic/tests/test_node_runtime_seturl.py

```
@@ -0,0 +1,1498 @@
+Tests for setURL transaction runtime classes including parser, planners,
+execution engine, rollback engine, cache manager, and coordinator.
+"""
+
+import base64
+import logging
+import threading
+from typing import NoReturn
+from unittest.mock import MagicMock
+
+import pytest
+from pytest import LogCaptureFixture
```

Ruff (0.15.6)

[warning] Incorrect import of `pytest`; use `import pytest` instead

(PT013)